semana10/projeto-labex/labex/src/assets/components/ApplicationFormPage.js

```diff
@@ -0,0 +1,146 @@
+import React, {useEffect, useState} from 'react'
+import TextField from '@material-ui/core/TextField';
+import Select from '@material-ui/core/Select'
+import Button from '@material-ui/core/Button';
+import { styled } from '@material-ui/styles'
+import Grid from '@material-ui/core/Grid'
+import { InputLabel, FormControl } from '@material-ui/core';
+import axios from "axios"
+import { baseUrl } from "../constants/axiosConstants"
+import useForm from '../hooks/useForm'
+import { useParams, useHistory } from "react-router-dom";
+import { goToTripPage } from '../router/goToPages';
+
+const FormDiv = styled(Grid)({
+    display: "grid",
+    gap: "1em",
+    alignItems: "center",
+    justifyItems: "center",
+    width: "100vw",
+    marginBottom: "50px"
+})
+
+const InputField = styled(TextField)({
+    width: "30vw",
+})
+
+const NamesInputField = styled(TextField)({
+    width: "30vw",
+})
+
+const DescriptionInputField = styled(TextField)({
+    width: "30vw",
+})
+
+const SelectOption = styled(Select)({
+    width: "30vw",
+})
+
+export default function ApplicationFormPage() {
+    const params = useParams();
+    const history = useHistory()
+    const { form, onChange, resetState } = useForm({
+        name: "",
+        age: "",
+        profession: "",
+        applicationText: "",
+        // country: "",
+      });
+    
+    const handleInputChange = (event) => {
+       
+        const { name, value } = event.target;
+
+        onChange(name, value);
+      };
+    
+    const handleSubmittion = () => {    
+        const body = {
+            name: form.name,
+            age: form.age,
+            applicationText: form.applicationText,
+            profession: form.profession,
+            country: selectedCountry
+        }
```

Comment:
ao invés de cria um body, acho que seria possível usa o próprio objeto form que já possui a mesma estrutura

Reply:
bom até poderia, mas to acostumada com body sla